impacket/smbserver.py

```diff
@@ -2497,7 +2497,7 @@ def smbComSessionSetupAndX(connId, smbServer, SMBCommand, recvPacket):
                 jtr_dump_path = smbServer.getJTRdumpPath()
                 ntlm_hash_data = outputToJohnFormat( b'', sessionSetupData['Account'], sessionSetupData['PrimaryDomain'], sessionSetupData['AnsiPwd'], sessionSetupData['UnicodePwd'] )
                 smbServer.log(ntlm_hash_data['hash_string'])
-                if jtr_dump_path is not '':
+                if jtr_dump_path != '':
```

Comment:
Referring to [PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) "For sequences, (strings, lists, tuples), use the fact that empty sequences are false":
```suggestion
                if jtr_dump_path:
```